sbt-dotty/src/dotty/tools/sbtplugin/DottyIDEPlugin.scala

```diff
@@ -89,40 +168,40 @@ object DottyIDEPlugin extends AutoPlugin {
   )
 
   override def buildSettings: Seq[Setting[_]] = Seq(
-    configureIDE := {
-      val log = streams.value.log
-
-      val configs0 = state.flatMap(s =>
-        inAllDottyConfigurations(projectConfig, s)
-      ).value.flatten
-      // Drop configurations who do not define their own sources, but just
-      // inherit their sources from some other configuration.
-      val configs = distinctBy(configs0)(_.sourceDirectories.deep)
-
-      if (configs.isEmpty) {
-        log.error("No Dotty project detected")
-      } else {
-        // If different versions of Dotty are used by subprojects, choose the latest one
-        // FIXME: use a proper version number Ordering that knows that "0.1.1-M1" < "0.1.1"
-        val ideVersion = configs.map(_.compilerVersion).sorted.last
+    configureIDE := Def.taskDyn {
```

Comment:
`configureIDE` should not be a `Task` anymore, if it's going to play with `State`s, create new ones and run tasks within those modified `State`s. It should definitely be a `Command`. Otherwise you expose yourself to race conditions if `configureIDE` is run in parallel with other tasks.